File: app/src/main/java/com/example/localai/model/ModelInfo.java
package com.example.localai.model;

import java.util.Arrays;
import java.util.List;

/** 模拟市场中的模型条目（P1 仅前端演示，不接入真实 Manifest）。 */
public class ModelInfo {

    public static final String COMPAT_RECOMMENDED = "RECOMMENDED";
    public static final String COMPAT_RUNNABLE = "RUNNABLE";
    public static final String COMPAT_HIGH_LOAD = "HIGH_LOAD";
    public static final String COMPAT_UNSUPPORTED = "UNSUPPORTED";

    public static final String TASK_TEXT = "TEXT";
    public static final String TASK_CODE = "CODE";

    public final String id;
    public final String name;
    public final String publisher;
    public final String paramsLabel;
    public final double paramsB;
    public final String quant;
    public final String sizeLabel;
    public final long sizeBytes;
    public final String contextLabel;
    public final String task;
    public final List<String> langs;
    public final String license;
    public final String desc;
    public final String compat;
    public final String compatReason;
    public final int ttftMs;
    public final double tps;
    public final int memMb;
    public final String updated;
    public final int gradIndex;
    public boolean installed;

    public ModelInfo(String id, String name, String publisher, String paramsLabel, double paramsB,
                     String quant, String sizeLabel, long sizeBytes, String contextLabel,
                     String task, List<String> langs, String license, String desc,
                     String compat, String compatReason, int ttftMs, double tps, int memMb,
                     String updated, int gradIndex, boolean installed) {
        this.id = id;
        this.name = name;
        this.publisher = publisher;
        this.paramsLabel = paramsLabel;
        this.paramsB = paramsB;
        this.quant = quant;
        this.sizeLabel = sizeLabel;
        this.sizeBytes = sizeBytes;
        this.contextLabel = contextLabel;
        this.task = task;
        this.langs = langs;
        this.license = license;
        this.desc = desc;
        this.compat = compat;
        this.compatReason = compatReason;
        this.ttftMs = ttftMs;
        this.tps = tps;
        this.memMb = memMb;
        this.updated = updated;
        this.gradIndex = gradIndex;
        this.installed = installed;
    }

    public char letter() {
        return Character.toUpperCase(name.charAt(0));
    }

    public static List<String> langs(String... values) {
        return Arrays.asList(values);
    }
}

File: app/src/main/kotlin/com/example/localai/model/ModelInfo.kt
package com.example.localai.model

import java.util.Arrays

/**
 * 市场/演示模型条目（P1 前端演示 + P4 起由真实目录映射填充）。
 * 含可变字段（installed 与 7 个 P4 扩展字段），故使用普通类 + @JvmField 保留与 Java 版相同的
 * 字段访问与对象同一性语义；license / quant 来自未做空值兜底的 manifest 字段，故保持可空。
 */
class ModelInfo(
    @JvmField val id: String,
    @JvmField val name: String,
    @JvmField val publisher: String,
    @JvmField val paramsLabel: String,
    @JvmField val paramsB: Double,
    @JvmField val quant: String?,
    @JvmField val sizeLabel: String,
    @JvmField val sizeBytes: Long,
    @JvmField val contextLabel: String,
    @JvmField val task: String,
    @JvmField val langs: List<String>,
    @JvmField val license: String?,
    @JvmField val desc: String,
    @JvmField val compat: String,
    @JvmField val compatReason: String,
    @JvmField val ttftMs: Int,
    @JvmField val tps: Double,
    @JvmField val memMb: Int,
    @JvmField val updated: String,
    @JvmField val gradIndex: Int,
    @JvmField var installed: Boolean
) {
    @JvmField var weightStatus: String? = null   // "approved" | "demo"
    @JvmField var isDemo: Boolean = false
    @JvmField var sourceUrl: String? = null
    @JvmField var licenseUrl: String? = null
    @JvmField var chatTemplate: String? = null
    @JvmField var minAndroidApi: Int = 0
    @JvmField var estimatedPeakMb: Int = 0

    fun letter(): Char = Character.toUpperCase(name[0])

    companion object {
        @JvmField val COMPAT_RECOMMENDED = "RECOMMENDED"
        @JvmField val COMPAT_RUNNABLE = "RUNNABLE"
        @JvmField val COMPAT_HIGH_LOAD = "HIGH_LOAD"
        @JvmField val COMPAT_UNSUPPORTED = "UNSUPPORTED"

        @JvmField val TASK_TEXT = "TEXT"
        @JvmField val TASK_CODE = "CODE"

        @JvmStatic
        fun langs(vararg values: String): List<String> = Arrays.asList(*values)
    }
}
